bool leftContact
float32 leftWeight
bool rightContact
float32 rightWeight
